suite-native/message-system/src/components/FeatureMessageScreen.tsx

```diff
@@ -0,0 +1,46 @@
+import { useDispatch } from 'react-redux';
+
+import { A, G } from '@mobily/ts-belt';
+
+import { Message } from '@suite-common/suite-types';
+import { messageSystemActions } from '@suite-common/message-system';
+import { useTranslate } from '@suite-native/intl';
+
+import { MessageScreen } from './MessageScreen';
+
+type FeatureMessageScreenProps = {
+    message: Message;
+};
+
+export const FeatureMessageScreen = ({ message }: FeatureMessageScreenProps) => {
+    const dispatch = useDispatch();
+    const { translate } = useTranslate();
+    const { id: messageId, dismissible: isDismissable, category, feature } = message;
+
+    const isKillswitch =
+        feature && A.isNotEmpty(feature?.filter(item => item.domain === 'killswitch' && item.flag));
+
+    const handleDismiss = () => {
+        if (!isDismissable) return;
+
+        const categories = G.isArray(category) ? category : [category];
+        categories.forEach(item => {
+            dispatch(
+                messageSystemActions.dismissMessage({
+                    id: messageId,
+                    category: item,
+                }),
+            );
+        });
+    };
+
+    return (
+        <MessageScreen
+            message={message}
+            defaultTitle={isKillswitch ? translate('messages.killswitch.title') : undefined}
```

Comment:
Why need to be passed the `defaultTitle`, `defaultContent` and `secondaryButtonLabel` sent as a prop? 

You are sending the message as a prop already, so you can figure out if it is or isn't killerSwitch inside of the `MessageScreen` and based on that figure out what should be the default messages.

Reply:
9860215781